js/enable-select2.js

```diff
@@ -13,28 +15,43 @@ CRM.$(function() {
    * We're debouncing the callback to avoid calling the plugin multiple times
    * during DOM changes
    */
-  var observer = new MutationObserver(debounce(function() {
+  var observer = new MutationObserver(debounce(function () {
     CRM.$('select:visible:not(.no-select2):not(.crm-form-multiselect)')
-      .select2({
+    .each(function () {
+      var select = CRM.$(this);
+      var hasNoSelect2Parent = select.closest('.no-select2').length;
```

Comment:
Not sure there is a need to assign a var, it is used in only one place, you can directly do 

```js
if (select.closest('.no-select2').length) { // this is not that long
  // ...
}
```

Reply:
the var is there more for an explanation (`hasNoSelect2parent`) rather than making things short.

Reply:
Well, I started searching where this var is used. If you think it provides better readability - I am not going to push it.